pkg/kubelet/util.go

```diff
@@ -118,3 +127,56 @@ func allowHostIPC(pod *api.Pod) (bool, error) {
 	}
 	return false, nil
 }
+
+type ContainerStatusbyCreated []*kubecontainer.ContainerStatus
+
+func (a ContainerStatusbyCreated) Len() int           { return len(a) }
+func (a ContainerStatusbyCreated) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ContainerStatusbyCreated) Less(i, j int) bool { return a[i].CreatedAt.After(a[j].CreatedAt) }
+
+// getContainersToDeleteInPod returns the exited containers in a pod whose name matches the name inferred from exitedContainerID, ordered by the creation time from the latest to the earliest.
+func getContainersToDeleteInPod(exitedContainerID string, podStatus *kubecontainer.PodStatus, maxDeadContainerInPod int) ContainerStatusbyCreated {
+	var matchedContainer *kubecontainer.ContainerStatus
+	var exitedContainers []*kubecontainer.ContainerStatus
+	// Find all exited containers in the pod
+	for _, containerStatus := range podStatus.ContainerStatuses {
+		if containerStatus.State != kubecontainer.ContainerStateExited {
+			continue
+		}
+		if containerStatus.ID.ID == exitedContainerID {
+			matchedContainer = containerStatus
+		}
+		exitedContainers = append(exitedContainers, containerStatus)
+	}
+	if matchedContainer == nil {
+		glog.Warningf("Container %q not found in pod's exited containers", exitedContainerID)
+		return ContainerStatusbyCreated{}
```

Comment:
We need to get the name of the exited container to find all the exited containers with the same name. This is a case that should never happen. (If it happens, it means pod cache is not updated properly.)